docs/src/manual/nested_autodiff.md

```diff
@@ -106,6 +106,16 @@ nothing; # hide
 That's pretty good, of course you will have some error from the finite differences
 calculation.
 
+!!! tip
+
+    Notice that in this example the Jacobian `J` consists on the full matrix of derivatives of `smodel` with respect
+    the different inputs in `x`. In order to avoid the unnecessary calculation of zero entries of the Jacobian, you can 
+    instead use [`batched_jacobian`](@ref) to parse the calculation of the Jacobian per each single input. 
+    ```@example nested_ad
+    J = batched_jacobian(smodel, AutoForwardDiff(), x)
```

Comment:
make it into a full function like `loss_function1`. Then if you add the checks with `@assert ... # hide` those will ensure that the computation is correct and won't be exposed in the docs

Reply:
I will do it! But please take a look at my following comment.